buildSrc/src/main/groovy/com/sun/javafx/gradle/NativeCompileTask.groovy

```diff
@@ -38,9 +39,9 @@ import java.util.concurrent.Executors
 import java.util.concurrent.Future
 
 class NativeCompileTask extends DefaultTask {
-    @Optional String matches; // regex for matching input files
-    List<String> params = new ArrayList<String>();
-    List sourceRoots = new ArrayList();
+    @Optional @Input String matches; // regex for matching input files
+    @Input List<String> params = new ArrayList<String>();
+    @Input List sourceRoots = new ArrayList();
```

Comment:
Shouldn't this be `@Internal`?:

1. `updateFiles()` uses `sourceRoots` and maps that into the `allFiles` field, which has already `@InputFiles`. If the `sourceRoots` is changed in a way that does not affect `allFiles`, this task can remain `UP_TO_DATE`.

Reply:
Perhaps, but I wouldn't want to make this change as part of this bug fix. I'd rather not have to do the analysis of whether there is any case where `sourceRoots` can change in a way that wouldn't affect the outputs. If not, then this is only a theoretical concern, and if so, it seems better to err on the side of correctness unless you can prove that all similar cases won't affect the outputs.